Sources/PointFree/Account/Change.swift

```diff
@@ -0,0 +1,392 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+// MARK: Middleware
+
+let subscriptionChangeShowResponse =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< fetchSeatsTaken
+    <<< requireStripeSubscription
+    <<< requireActiveSubscription
+    <| writeStatus(.ok)
+    >-> map(lower)
+    >>> respond(
+      view: subscriptionChangeShowView,
+      layoutData: { subscription, currentUser, seatsTaken in
+        SimplePageLayoutData(
+          currentSubscriptionStatus: subscription.status,
+          currentUser: currentUser,
+          data: (subscription, currentUser, seatsTaken),
+          extraStyles: extraStyles,
+          title: "Modify subscription"
+        )
+    }
+)
+
+let subscriptionChangeMiddleware =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< filterMap(
+      require2 >>> pure,
+      or: redirect(
+        to: .account(.subscription(.change(.show))),
+        headersMiddleware: flash(
+          .error,
+          "Invalid subscription data. Please try again or contact <support@pointfree.co>."
+        )
+      )
+    )
+    <<< fetchSeatsTaken
+    <<< requireStripeSubscription
+    <<< requireActiveSubscription
+    <<< requireValidSeating
+    <| map(lower)
+    >>> subscriptionChange
+
+private func subscriptionChange(_ conn: Conn<StatusLineOpen, (Stripe.Subscription, Database.User, Int, Pricing)>)
+  -> IO<Conn<ResponseEnded, Data>> {
+
+    let (currentSubscription, user, _, newPricing) = conn.data
+
+    let newPrice = defaultPricing(for: newPricing.lane, billing: newPricing.billing) * 100
+    let currentPrice = currentSubscription.plan.amount.rawValue
+
+    let prorate = newPrice > currentPrice
+
+    return AppEnvironment.current.stripe.updateSubscription(currentSubscription, newPricing.plan, newPricing.quantity, prorate)
+      .run
+      .flatMap(
+        either(
+          const(
+            conn |> redirect(
+              to: .account(.subscription(.change(.show))),
+              headersMiddleware: flash(
+                .error,
+                """
+                We couldn’t modify your subscription at this time. Please try again or contact
+                <support@pointfree.co>.
+                """
+              )
+            )
+          )
+        ) { _ in
+          // TODO: Send email?
+
+          return conn |> redirect(
+            to: .account(.index),
+            headersMiddleware: flash(.notice, "We’ve modified your subscription.")
+          )
+        }
+    )
+}
+
+func requireActiveSubscription<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data> {
```

Comment:
it doesn't look like `User` is used in this middleware so could it just be `T2<Subscription, Z>`?